Split the non-tabular node to generate one item per resource

Currently, we are copping across non-tabular resources, without any modification to dp.json. We need to change this behaviour as in some cases we have to reuse and add it as a new resource to the dp.json. `add_resource` processor searches the resource in dp.json with its name and as non-tabular ones are copied across, resource names are copied from originals. So we are getting `AssertionError: Failed to find a resource with the index or name matching 'non-tabular'` as a planner is trying to add resource using artefact name `(add_resource, {resource: resource_info[required_artifact.resource_name]}...)`

# Acceptance Criteria

* [ ] generate one item per resource for non-tabular ones as well 

# Tasks 

*  [ ] refactor `NonTabularProcessingNode` in basic_nodes.py to yield artifact with correct resource names `{resource name}_non_tabular`

--- a/datapackage_pipelines_assembler/nodes/basic_nodes.py
+++ b/datapackage_pipelines_assembler/nodes/basic_nodes.py
@@ -49,10 +49,25 @@ class NonTabularProcessingNode(BaseProcessingNode):
         super(NonTabularProcessingNode, self).__init__(available_artifacts, outputs)
 
     def get_artifacts(self):
-        output = ProcessingArtifact(
-            None, 'non-tabular',
-            [], self.available_artifacts,
-            [],
-            False
-        )
-        yield output
+        for artifact in self.available_artifacts:
+            datahub_type = 'derived/non-tabular'
+            resource_name = artifact.resource_name + '_{}'.format('non-tabular')
+            output = ProcessingArtifact(
+                datahub_type, resource_name,
+                [], [artifact],
+                [('assembler.update_resource',
+                  {
+                      'name': artifact.resource_name,
+                      'update': {
+                          'name': resource_name,
+                          'datahub': {
+                            'type': datahub_type,
+                            'derivedFrom': [
+                                artifact.resource_name
+                            ]
+                          }
+                      }
+                  })],
+                False
+            )
+            yield output

